### PR TITLE
add llamaindex + groq cookbook

### PR DESCRIPTION
**Pull request by jerryjliu**
same as #502 but with groq

### CODE REVIEW

**Comment by jerryjliu** [comment]
> Thanks @jerryjliu for the PR can we pls add it here, under this folder, https://github.com/meta-llama/llama-recipes/tree/main/recipes/llama_api_providers

lol my bad! moving it now

**Comment by jerryjliu** [comment]
@HamidShojanazeri check now

File: recipes/llama_api_providers/llama3_cookbook_groq.ipynb [hunk 1]
```diff
@@ -0,0 +1,1472 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d5256970-eba4-499a-b438-8766a290a61a",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from llama_index.llms.groq import Groq\n",
+    "\n",
+    "llm = Groq(model=\"llama3-8b-8192\")\n",
```
Comment (jeffxtang) [inline recipes/llama_api_providers/llama3_cookbook_groq.ipynb:76]: should add 
os.environ["GROQ_API_KEY"] = GROQ_API_KEY
or change to
llm = Groq(model="llama3-8b-8192", api_key=GROQ_API_TOKEN)

File: recipes/llama_api_providers/llama3_cookbook_groq.ipynb [hunk 2]
```diff
@@ -0,0 +1,1472 @@
...
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9edee491-05f8-4fbb-9394-baa82f1e5087",
+   "metadata": {},
+   "source": [
+    "### Load Data\n",
+    "\n",
+    "We load data using LlamaParse by default, but you can also choose to opt for our free pypdf reader (in SimpleDirectoryReader by default) if you don't have an account."
```
Comment (jeffxtang) [inline recipes/llama_api_providers/llama3_cookbook_groq.ipynb:155]: Please add a note saying how to sign up to get a key. or uncomment the free one and comment the LlamaParse one. Also, is pypdf good enough? When should one use it and when LlamaParse? @jerryjliu

**Review by jeffxtang — COMMENTED** [review]
Thanks Jerry. Found a couple of errors while running the notebook. @jerryjliu

File: recipes/llama_api_providers/llama3_cookbook_groq.ipynb [hunk 3]
```diff
@@ -0,0 +1,1491 @@
...
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "21e05b95-95cb-45f3-8fbe-d0b54a9ba988",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# from llama_index.core.query_engine import SubQuestionQueryEngine\n",
```
Comment (jeffxtang) [inline recipes/llama_api_providers/llama3_cookbook_groq.ipynb:586]: remove these commented lines? or explain or uncomment?

Comment (jerryjliu) [reply recipes/llama_api_providers/llama3_cookbook_groq.ipynb]: @jeffxtang just tested again. for some reason i can't get sub-question to work on groq 70b, so i took the sub-question section out

File: recipes/llama_api_providers/llama3_cookbook_groq.ipynb [hunk 4]
```diff
@@ -0,0 +1,1422 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "0e241fe9-f390-4be5-b3c4-da4f56db01ef",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "drake_tool = QueryEngineTool(\n",
+    "    drake_index.as_query_engine(),\n",
```
Comment (jeffxtang) [inline recipes/llama_api_providers/llama3_cookbook_groq.ipynb:1332]: image

got an error @jerryjliu

**Review by jeffxtang — APPROVED** [review]

**Pull request merged**